llvm/cmake/modules/HandleLLVMOptions.cmake

```diff
@@ -36,6 +36,21 @@ string(TOUPPER "${LLVM_ENABLE_LTO}" uppercase_LLVM_ENABLE_LTO)
 # The following only works with the Ninja generator in CMake >= 3.0.
 set(LLVM_PARALLEL_COMPILE_JOBS "" CACHE STRING
   "Define the maximum number of concurrent compilation jobs (Ninja only).")
+if(LLVM_RAM_PER_COMPILE_JOB OR LLVM_RAM_PER_LINK_JOB)
+  cmake_host_system_information(RESULT available_physical_memory QUERY AVAILABLE_PHYSICAL_MEMORY)
+  cmake_host_system_information(RESULT number_of_logical_cores QUERY NUMBER_OF_LOGICAL_CORES)
+endif()
+if(LLVM_RAM_PER_COMPILE_JOB)
+  math(EXPR memory_available_jobs "${available_physical_memory} / ${LLVM_RAM_PER_COMPILE_JOB}" OUTPUT_FORMAT DECIMAL)
```

Comment:
`memory_available_jobs` is a very confusing variable name.  What do you think of something like `max_parallelism` or `possible_threads`?

Reply:
Sure I can change it, is `jobs_with_sufficient_memory` fine for you ?
`max_parallelism` would seem imprecise, I would expect `max_parallelism` to be equal to `LLVM_PARALLEL_COMPILE_JOBS` so the result of last branch. Same for `possible_threads`, even `possible_jobs` because all around is talked about jobs.

Reply:
btw I hope its fine Iam rebasing this, seems like the downside is that github is losing track of conversation.

Reply:
Sure, `jobs_with_sufficient_memory` should be fine.